Mod does not boot without Fabric API

### minecraft version

another version

### mod version

1.1.1

### mod loader

fabric

### what went wrong?

Using fabric 0.17.3, minecraft 1.21.9 the client does not boot when only the following mods are installed:
placeholder-api-2.8.0+1.21.9
polymer-bundled-0.14.2+1.21.9
styled_renaming-1.1.1+1.21.5

These mods crash with this message

---- Minecraft Crash Report ----
// You're mean.
Time: 2025-10-22 23:55:07
Description: Initializing game
java.lang.RuntimeException: Could not execute entrypoint stage 'main' due to errors, provided by 'styled_renaming' at 'dev.rosebud.styled_renaming.StyledRenaming'!
at net.fabricmc.loader.impl.FabricLoaderImpl.lambda$invokeEntrypoints$2(FabricLoaderImpl.java:409)
at net.fabricmc.loader.impl.util.ExceptionUtil.gatherExceptions(ExceptionUtil.java:33)
at net.fabricmc.loader.impl.FabricLoaderImpl.invokeEntrypoints(FabricLoaderImpl.java:407)
at net.fabricmc.loader.impl.game.minecraft.Hooks.startClient(Hooks.java:52)
at knot//net.minecraft.class_310.<init>(class_310.java:483)
at knot//net.minecraft.client.main.Main.main(Main.java:234)
at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:506)
at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:72)
at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(Unknown Source)
at java.base/java.lang.reflect.Method.invoke(Unknown Source)
at com.modrinth.theseus.MinecraftLaunch.relaunch(MinecraftLaunch.java:63)
at com.modrinth.theseus.MinecraftLaunch.main(MinecraftLaunch.java:28)
Caused by: java.lang.ExceptionInInitializerError
at java.base/java.lang.Class.forName0(Native Method)
at java.base/java.lang.Class.forName(Unknown Source)
at java.base/java.lang.Class.forName(Unknown Source)
at net.fabricmc.loader.impl.util.DefaultLanguageAdapter.create(DefaultLanguageAdapter.java:50)
at net.fabricmc.loader.impl.entrypoint.EntrypointStorage$NewEntry.getOrCreate(EntrypointStorage.java:124)
at net.fabricmc.loader.impl.entrypoint.EntrypointContainerImpl.getEntrypoint(EntrypointContainerImpl.java:53)
at net.fabricmc.loader.impl.FabricLoaderImpl.invokeEntrypoints(FabricLoaderImpl.java:405)
... 10 more
Caused by: java.lang.IllegalStateException: Registry is already frozen (trying to add key ResourceKey[minecraft:data_component_type / styled_renaming:raw_name])
at knot//net.minecraft.class_2370.method_40274(class_2370.java:95)
at knot//net.minecraft.class_2370.method_10272(class_2370.java:101)
at knot//net.minecraft.class_2378.method_39197(class_2378.java:126)
at knot//net.minecraft.class_2378.method_10230(class_2378.java:122)
at knot//dev.rosebud.styled_renaming.StyledRenaming.<clinit>(StyledRenaming.java:18)
... 17 more

Adding fabric-api-0.134.0+1.21.9 allows to boot and the mod works perfectly.

### log file

_No response_

--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -127,7 +127,7 @@ publishMods {
 
         projectDescription = rootProject.file("README.md").readText()
 
-        requires("polymer", "placeholder-api")
+        requires("polymer", "placeholder-api", "fabric-api")
     }
 }
 

--- a/src/main/resources/fabric.mod.json
+++ b/src/main/resources/fabric.mod.json
@@ -18,7 +18,8 @@
     "depends": {
         "minecraft": ">=1.21",
         "polymer-core": ">=0.9.0",
-        "placeholder-api": ">=2.4.0"
+        "placeholder-api": ">=2.4.0",
+        "fabric-registry-sync-v0": "*"
     },
     "mixins": ["${id}.mixins.json"]
 }

